nullaway/src/test/java/com/uber/nullaway/TypeUseAnnotationsTests.java

```diff
@@ -189,21 +193,148 @@ public void typeUseAnnotationOnInnerMultiLevel() {
         .addSourceLines(
             "Test.java",
             "package com.uber;",
-            "import java.util.Set;",
             "import org.checkerframework.checker.nullness.qual.Nullable;",
             "class A { class B { class C {} } }",
             "class Test {",
-            "  // At some point, we should not treat foo1 or foo2 as @Nullable.",
-            "  // For now we do, for ease of compatibility.",
-            "  // TODO: Fix this as part of https://github.com/uber/NullAway/issues/708",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "  @Nullable A.B.C foo1 = null;",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "  A.@Nullable B.C foo2 = null;",
+            "  A.B.@Nullable C foo3 = null;",
+            "  // BUG: Diagnostic contains: Type-use nullability annotations should be applied on inner class",
+            "    @Nullable A.B foo4 = null;",
+            "  // BUG: Diagnostic contains: assigning @Nullable expression to @NonNull field",
+            "  A.B.@Nullable C [][] foo5 = null;",
```

Comment:
If we are handling multi-dim arrays can we add a test where the field is actually `@Nullable`?  I think you need to write `A.B.C @Nullable [][]`